### ISSUE TITLE
suggestion : Can I use my uv loop to create EventLoopSPtr ?

### ISSUE THREAD

**Issue by wolf289184291** [sewenew/redis-plus-plus#238]
In my project, libuv is integrated by core module(dll), and redis++ is the third party code,
My project is not supply libuv.h and libuv.lib to other dlls,
so could redis++ supplies a new interface, such as AttachUVLoop(loop*), to let code more better for some special users?
Thanks!

**Comment by sewenew** [comment]
Yes, there's a plan to support that, although not an interface like AttachUVLoop. Instead, I was plan to allow end user to create a EventLoop with a given uv_loop_t*. EventLoop won't free the memory of this loop pointer, instead, the end user is in change of destroying the uv_loop_t pointer. I'm not sure if this work for you, and the following is an example of how to use it:

```
// uv loop created by user.
auto *loop = new uv_loop_t;
uv_loop_init(loop);

auto event_loop = make_shared<EventLoop>(loop);   // in this case, EventLoop won't destroy the loop
auto redis = AsyncRedis(connection_opts, pool_opts, event_loop);

// Application code is in charge of destroying the loop.
uv_loop_close(loop);
delete loop;
```

If this works for you, I'll try to implement it ASAP.

Regards

**Comment by samantehrani** [comment]
Hello, I am working on a distributed ws project where I use redis as the application cache. I use usWebsocket for network processing which has it's own libuv event loop. Using redis-plus-plus client  I haven't been able to figure out an efficient way to synchronize network's event loop with redis client's event loop.
The interface you mentioned above could be a reasonable solution, just having a single libuv loop for processing both network and cache.

To give an example of the problem, imagine the following:
```
// new message from network that requires few serialized cache interactions
// that all block the network thread
NetworkLoop.onMessage() {
  cache.getSomeValueA().get();
  cache.getPutValueA().get();
  cache.getSomeValueB().get();
  cache.getPutValueB().get();
}
```
As an alternative, I can create a new thread for processing each message:
```
NetworkLoop.onMessage() {
  std::async([](){
  cache.getSomeValueA().get();
  cache.getPutValueA().get();
  cache.getSomeValueB().get();
  cache.getPutValueB().get();
  });
}
```
but this is not efficient creating one thread per incoming message. 

Let me know what you think or any idea on the complexity of the solution you mentioned above.

**Comment by sewenew** [comment]
The solution I mentioned above should not be a complex one. I'll try to implement it when I have time (maybe next week). Sorry, too busy these days...

Regards

**Comment by samantehrani** [comment]
Thanks a lot.

**Comment by mroland91** [comment]
Hi, I'd just like to bump this topic.
